src/main.rs

```diff
@@ -57,9 +57,9 @@ fn process_package(dir_output: &PathBuf, file: &PathBuf) -> Result<(), Error> {
     let output_tar_gz_path: PathBuf =
         Path::new(&dir_output).join(Path::new(&tar_gz_file_name.replace("TDR", "TST")));
     let uncompressed_folder_input_path: &PathBuf = &file.with_extension("").with_extension("");
-    let input_batch_reference: &str = uncompressed_folder_input_path
+    let input_batch_reference: String = uncompressed_folder_input_path
         .file_name()
-        .and_then(|name| name.to_str())
+        .and_then(|name| name.to_str().map(|name| name.replace("TRE-", "")))
```

Comment:
1. I'm not able to see what the result of `name.to_str()` (I assumed it's a String but could be a container) but can't you do `name.to_str().replace("TRE-", "")` instead?
2. From what I understand, there is `to_str` and `to_string` I think `to_string` might be newer; (is that's the case) should we use that instead?

Reply:
It's not a `String` it's an `OsStr`which doesn't have a direct `to_string` function. `to_str` returns an Option. I think the logic is that it can return different things depending on the underlying operating system or something like that. Anyway, it's an Option which is why we need the `map` call.